### PR TITLE
Postgres cleanup 1113

### PR DESCRIPTION
**Pull request by ianco**
Added storage_credentials to CLI wallet delete

Code cleanup of postgres wallet:

- Removed unnecessary error codes
- Removed (duplicated) language.rs module

Other code is all required:

utils.crypto.base64 = encoding and decoding of tags, see:
     - https://github.com/ianco/indy-sdk/blob/postgres_cleanup_1113/samples/storage/storage-postgres/src/postgres_wallet.rs#L853
     - https://github.com/ianco/indy-sdk/blob/postgres_cleanup_1113/samples/storage/storage-postgres/src/wallet_storage/language.rs#L35

utils.crypto.chacha20poly1305_ietf = required for const ENCRYPTED_KEY_LEN, used by EncryptedValue, used to decode wallet storage parameters
Code organization is based on whether code is specific to postgres storage or copied from libindy:
     - https://github.com/ianco/indy-sdk/blob/postgres_cleanup_1113/samples/storage/storage-postgres/src/wallet_storage/storage.rs#L16

- postgres_storage module is specific to postgres
- api, errors, utils and wallet_storage are all "cut & paste" copied from libindy

### CODE REVIEW

**Comment by vimmerru** [comment]
@ianco Thanks for supporting this. I will add comments with suggestions during my review.

**Comment by vimmerru** [comment]
1. I suggest to remove api/mod.rs and move ErrorCode to libindy/mod.rs

**Comment by vimmerru** [comment]
3. I see structure like

```
- src
  - lib.rs // here init method and all c handlers as private functions without prefixes like _create/_open
  - storage.rs // implementation of PostgressStorage and PostgressStorageType
  - wql // set of utils for wql parsing and sql generation that in the future can be shared as separate library
 - utils // only utils you really need
```

Note: We don't need WalletStorage and WalletStorageType trates here. 

Now you have:

- wallet_storage
- postgress_storage
- postgress_wallet

It is really confusing.

**Comment by vimmerru** [comment]
4. All type checks in c handlers should return IvalidState error as all strings will be pre-validated by libindy

**Comment by vimmerru** [comment]
5. I suggest to move root folder from samples to `experimental/plugins/postgress_storage`

**Comment by ianco** [comment]
Updated per @vimmerru feedback:

1.  Removed api/mod.rs and moved ErrorCode to libindy/mod.rs
2.  No changes to error code - future change
3.  Re-organized code structure to simplify
4.  C handlers return CommonInvalidState error
5.  Moved code to experimental/plugins/postgres_storage

Also have made the following changes:

- Added wallet_storage support to CLI wallet delete command
- Added wallet storage support to python getting_started
- Added bash scripts to postgres_storage module to illustrate calling CLI and python getting_started using the postgres wallet storage (note these are also used to test updates to the storage)

**Comment by ianco** [comment]
Cleanup is completed, still 2 outstanding issues, I've opened up separate JIRA items (IS-1114, IS-1129)

Included updates to python getting_started.py to support running with postgres plug-in, and added documentation (mostly copied from the HIPE)

**Comment by vimmerru** [comment]
@ianco Thanks for this contribution, will look ASAP. Sorry for late response, just returned from the holidays.

**Comment by ianco** [comment]
There are conflicts due to this commit - https://github.com/hyperledger/indy-sdk/commit/5f99530149c488c4ee3f3fa2e37196461e9e4b86

I re-organized the code based on your suggestions (moved from "samples" to "experimental") but some of the "samples" files were updated with new error handling.

**Comment by ianco** [comment]
@vimmerru  any update on this?  I took a quick look and the code in indy-sdk currently doesn't compile so looks like there is a merge currently in progress (?)  Thanks.

**Comment by vimmerru** [comment]
@ianco current master now uses new error handling approach based on ErrorKind pattern and failure library. Most probably the conflict is related to these changes. Do you need help with merging?

**Comment by ianco** [comment]
Hi @vimmerru , there are 2 issues:

1. Code in indy-sdk master branch does not compile
2. There are conflicts between the code in master branch (changes to code in the old "samples" dir vs the code moved to "experimental"

I wasn't sure if there was work still in progress or if it was up to me to fix these compile errors/merge conflicts.  Sounds like it's on me?

**Comment by ianco** [comment]
@vimmerru  I fixed the merge conflicts and verified everything compiles and the tests run.  I backed out the (partial) changes relating to error code as it looks like they were committed into the postgres code by mistake.

Note I have a separate ticket (IS-1129) for applying these error code changes to the postgres plug-in.

File: cli/src/commands/wallet.rs [hunk 1]
```diff
@@ -301,11 +302,21 @@ pub mod delete_command {
...
             .map_err(|_| println_err!("Wallet \"{}\" isn't attached to CLI", id))?;
 
-        let credentials: String = json!({ "key": key.clone(), "key_derivation_method": map_key_derivation_method(key_derivation_method)? }).to_string();
+        let credentials = {
+            let mut json = JSONMap::new();
+
+            json.insert("key".to_string(), serde_json::Value::String(key.to_string()));
+            json.insert("key_derivation_method".to_string(), serde_json::Value::String(map_key_derivation_method(key_derivation_method)?.to_string()));
+
+            storage_credentials.map(|creds| json.insert("storage_credentials".to_string(), creds));
```
Comment (vimmerru) [inline cli/src/commands/wallet.rs]: It isn't good idea to use "map" for side effect action. Simple "if let" will be better. Anyway we will merge it fix during merging changes from Wrapper Errors PR.

**Pull request merged**